frodo.js

```diff
@@ -1,12 +1,12 @@
 /**
```

Comment:
Well done with the efficient error handling, it covers all the possible scenarios.